tests/test_types.py

```diff
@@ -6129,3 +6133,38 @@ class MyModel(BaseModel):
 
     round_trip_value = json.loads(MyModel(val=True).model_dump_json())['val']
     assert round_trip_value is True, round_trip_value
+
+
+def test_on_error_omit() -> None:
+    OmittableInt = OnErrorOmit[int]
+
+    class MyTypedDict(TypedDict):
+        a: NotRequired[OmittableInt]
+        b: NotRequired[OmittableInt]
+
+    class Model(BaseModel):
+        a_list: list[OmittableInt]
+        a_dict: dict[OmittableInt, OmittableInt]
+        a_typed_dict: MyTypedDict
+
+    actual = Model(
+        a_list=[1, 2, 'a', 3],
+        a_dict={1: 1, 2: 2, 'a': 'a', 'b': 0, 3: 'c', 4: 4},
+        a_typed_dict=MyTypedDict(a=1, b='xyz'),  # type: ignore
+    )
+
+    expected = Model(a_list=[1, 2, 3], a_dict={1: 1, 2: 2, 4: 4}, a_typed_dict=MyTypedDict(a=1))
+
+    assert actual == expected
+
```

Comment:
can we add a quick test for list, dict key and dict value?

Reply:
That's already in this test, I'm not sure what you're asking for